ikarus/controlroutines/loadcontrol.hh

```diff
@@ -35,28 +32,32 @@ namespace Ikarus {
           "The last parameter (load factor) must be assignable and incrementable with a double!");
     }
 
-    LoadControlInformation run() {
-      LoadControlInformation info({false});
+    ControlInformation run() {
+      ControlInformation info({false});
       auto& nonOp = nonLinearSolver->nonLinearOperator();
       this->notify(ControlMessages::CONTROL_STARTED);
       auto& loadParameter = nonOp.lastParameter();
 
       loadParameter = 0.0;
-      this->notify(ControlMessages::STEP_STARTED);
+      this->notify(ControlMessages::STEP_STARTED, 0, stepSize_);
       auto solverInfo = nonLinearSolver->solve();
+      info.solverInfos.push_back(solverInfo);
+      info.totalIterations += solverInfo.iterations;
       if (not solverInfo.success) return info;
       this->notify(ControlMessages::SOLUTION_CHANGED);
       this->notify(ControlMessages::STEP_ENDED);
 
       for (int ls = 0; ls < loadSteps_; ++ls) {
-        this->notify(ControlMessages::STEP_STARTED);
+        this->notify(ControlMessages::STEP_STARTED, ls, stepSize_);
         loadParameter += stepSize_;
         solverInfo = nonLinearSolver->solve();
+        info.solverInfos.push_back(solverInfo);
+        info.totalIterations += solverInfo.iterations;
         if (not solverInfo.success) return info;
         this->notify(ControlMessages::SOLUTION_CHANGED);
         this->notify(ControlMessages::STEP_ENDED);
       }
-      this->notify(ControlMessages::CONTROL_ENDED);
+      this->notify(ControlMessages::CONTROL_ENDED, info.totalIterations, "Load Control Method");
```

Comment:
Why is here somehow the name of the controlroutine passed? Is this just for printing the name at the end?
If yes, this communication should happen somewhere else (at `CONTROL_STARTED` or creating a "name" method for each Controlroutine